File: java/src/test/java/fr/sdecout/handson/rest/library/LibraryIdTest.java
package fr.sdecout.handson.rest.library;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;

public class LibraryIdTest {

    @Test
    void should_fail_to_initialize_from_blank_value() {
        assertThatIllegalArgumentException()
                .isThrownBy(() -> new LibraryId(""))
                .withMessage("Library ID must not be blank");
    }

    @Test
    void should_fail_to_initialize_from_value_with_more_than_36_characters() {
        var value = "1111111111111111111111111111111111111";
        assertThat(value).hasSize(37);

        assertThatIllegalArgumentException()
                .isThrownBy(() -> new LibraryId(value))
            .withMessage("Library ID must not have more than 36 characters");
    }

    @Test
    void should_display_as_string() {
        var value = "111111111111111111111111111111111111";
        assertThat(value).hasSize(36);

        assertThat(new LibraryId(value)).hasToString(value);
    }

}

File: kotlin/src/test/kotlin/fr/sdecout/handson/rest/library/LibraryIdTest.kt
package fr.sdecout.handson.rest.library

import fr.sdecout.handson.rest.library.LibraryId
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatIllegalArgumentException
import org.junit.jupiter.api.Test

class LibraryIdTest {

    @Test
    fun `should fail to initialize from blank value`() {
        assertThatIllegalArgumentException()
            .isThrownBy { LibraryId("") }
            .withMessage("Library ID must not be blank")
    }

    @Test
    fun `should fail to initialize from value with more than 36 characters`() {
        val value = "1111111111111111111111111111111111111"
        assertThat(value).hasSize(37)

        assertThatIllegalArgumentException()
            .isThrownBy { LibraryId(value) }
            .withMessage("Library ID must not have more than 36 characters")
    }

    @Test
    fun `should display as string`() {
        val value = "111111111111111111111111111111111111"
        assertThat(value).hasSize(36)

        assertThat(LibraryId(value)).hasToString(value)
    }

}
